apps/prairielearn/src/models/variant.ts

```diff
@@ -237,3 +237,19 @@ export async function selectAndAuthzVariant(options: {
 
   return variant;
 }
+
+/**
+ * Locks the variant before a grading operation can proceed. If the variant is
+ * associated to an assessment instance, lock the assessment instance instead.
+ * Assumes that the caller is already within a transaction.
+ */
+export async function lockVariant({ variant_id }: { variant_id: string }) {
+  const locked = await queryRow(
```

Comment:
I think the `queryRow` here is assuming that the variant exists; otherwise zero rows would be returned. Is that right? I'm OK with that, but it may be a little more defensive/explicit to use `queryOptionalRow` here.